evm_arithmetization/src/cpu/kernel/asm/core/call_gas.asm

```diff
@@ -71,7 +71,13 @@ global xfer_cost:
     %jump(after_xfer_cost)
 xfer_cost_nonzero:
     // stack: cost, is_call_or_staticcall, is_call_or_callcode, address, gas, kexit_info, value, retdest
-    %add_const(@GAS_CALLVALUE)
+    SWAP5
+    // stack: kexit_info, is_call_or_staticcall, is_call_or_callcode, address, gas, cost, value, retdest
+    PUSH @GAS_CALLVALUE
+    // stack: call_value_gas, kexit_info, is_call_or_staticcall, is_call_or_callcode, address, gas, cost, value, retdest
+    %charge_gas
+    // stack: kexit_info, is_call_or_staticcall, is_call_or_callcode, address, gas, cost, value, retdest
+    SWAP5
```

Comment:
Shouldn't we still update the `cost` variable to `cost+GAS_CALLVALUE` here to make sure `C_extra` is correct?

Reply:
Possibly? I was following my understanding of the pattern [here](https://github.com/0xPolygonZero/zk_evm/pull/124/files) in which I believe `cost` only reflects the gas cost which hasn't already been charged for?

Reply:
Ah no nevermind it's all good. LGTM 👍